netdata/netdata
# Fix time_t format

##### Summary
`time_t` type may be different on different platforms. We should cast it to a definite type before printing.

Fixes #11830

Thank you @WYC-2020!

##### Test Plan
The PR was tested on the `openwrt/openwrt` master branch.
1. Prepare an OpenWrt build for 32bit architecture (x86 generic). It's a very lo-o-ong and space-consuming process 😉.
```
git clone https://git.openwrt.org/openwrt/openwrt.git
./scripts/feeds update -a
./scripts/feeds install -a
make menuconfig
make -j8
```
2. Make changes to the netdata package
- `Makefile`
```
PKG_VERSION:=time_t-format
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/vlvkobal/netdata/tar.gz/$(PKG_VERSION)?
PKG_HASH:=4c9615cde21e427c0ab3d900a6acdb0ba7468b89507e8ab480e699f182b8d86a

DEPENDS:=+zlib +libuuid +libuv +libmnl +libjson-c +libopenssl

TARGET_LDFLAGS += -Wl,--gc-sections,-lssl,-lcrypto
```
- Remove patches for `index.html`
- Add a patch for `database/sqlite/sqlite_functions.h`
```
--- a/database/sqlite/sqlite_functions.h
+++ b/database/sqlite/sqlite_functions.h
@@ -3,6 +3,7 @@
 #ifndef NETDATA_SQLITE_FUNCTIONS_H
 #define NETDATA_SQLITE_FUNCTIONS_H

+#include <openssl/evp.h>
 #include "daemon/common.h"
 #include "sqlite3.h"

```
3. Rebuild the netdata package and make sure all the warnings for `time_t` are gone. 
```
make -j1 V=sc
```
